db/migrations/20220512132042_create_repository_configurations_table.up.sql

```diff
@@ -1,21 +1,91 @@
 BEGIN;
 
+--
+-- repositories
+--
+CREATE TABLE IF NOT EXISTS repositories (
+    uuid UUID UNIQUE NOT NULL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE,
+    updated_at TIMESTAMP WITH TIME ZONE,
+    url VARCHAR(255) NOT NULL,
+    last_read_time TIMESTAMP WITH TIME ZONE DEFAULT NULL,
+    last_read_error VARCHAR(255) DEFAULT NULL
+);
+
+ALTER TABLE repositories
+ADD CONSTRAINT repositories_unique_url UNIQUE (url);
+
+--
+-- repository_configurations
+--
 CREATE TABLE IF NOT EXISTS repository_configurations(
     uuid UUID UNIQUE NOT NULL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL,
+    updated_at TIMESTAMP WITH TIME ZONE NOT NULL,
+
     name VARCHAR (255) NOT NULL,
-    url VARCHAR (255) NOT NULL,
     versions VARCHAR (255)[],
     arch VARCHAR (255) NOT NULL,
     account_id VARCHAR (255) NOT NULL,
     org_id VARCHAR (255) NOT NULL,
-    created_at timestamp NOT NULL,
-    updated_at timestamp NOT NULL
+    repository_uuid UUID NOT NULL
     );
 
 ALTER TABLE repository_configurations
-ADD CONSTRAINT url_and_org_id_unique UNIQUE (url, org_id);
+ADD CONSTRAINT repo_and_org_id_unique UNIQUE (repository_uuid, org_id);
+
+ALTER TABLE repository_configurations
+ADD CONSTRAINT fk_repository
+FOREIGN KEY (repository_uuid)
+REFERENCES repositories(uuid)
+ON DELETE SET NULL;
 
 ALTER TABLE repository_configurations
 ADD CONSTRAINT name_and_org_id_unique UNIQUE (name, org_id);
 
+--
+-- rpm
+--
+
+CREATE TABLE IF NOT EXISTS rpms (
+    uuid UUID UNIQUE NOT NULL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE,
+    updated_at TIMESTAMP WITH TIME ZONE,
+
+    name VARCHAR(255) NOT NULL,
```

Comment:
There isn't actually a performance benefit for limiting this to 255:  https://dba.stackexchange.com/questions/20974/should-i-add-an-arbitrary-length-limit-to-varchar-columns

The default limit is 65K, we should probably leave it alone or set it to some really really high limit, like 2,000.  Since we're pulling this data in from other sources, the only reason to limit it would be to reduce malicious users that may just be trying to fill up space  (granted they could do that with just a large number of rpms i guess).  So maybe we shouldn't limit any fields

Reply:
Thanks! My postgres skills are rusted :cry:   ; I have seen gorm map strings to `TEXT` column type; is it ok to update column types to `TEXT`? should all the database fields be changed or only Rpm table?

Reply:
thanks! updated